src/core/controllers/base.go

```diff
@@ -108,8 +108,46 @@ func (cc *CommonController) Login() {
 	cc.PopulateUserSession(*user)
 }
 
-// LogOut Habor UI
+// LogOut Harbor UI
 func (cc *CommonController) LogOut() {
+	// redirect for OIDC logout, excludes the admin user.
+	securityCtx, ok := security.FromContext(cc.Context())
+	if !ok {
+		log.Error("Failed to get security context")
+		cc.CustomAbort(http.StatusInternalServerError, "Internal error.")
```

Comment:
Can we not have a precise error message, explaining what actually went wrong?

Reply:
Regarding the 500 error, we used to only provide more details in the log and not expose them further.

Reply:
does exposing the error, provide a threat, or might help the user to resolve the issue himself?